components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaProducer.java

```diff
@@ -214,15 +222,39 @@ public void remove() {
 
         ProducerRecord record;
         if (hasPartitionKey && hasMessageKey) {
-            record = new ProducerRecord(topic, partitionKey, key, value);
+            record = new ProducerRecord(topic, partitionKey, null, key, value, propagatedHeaders);
         } else if (hasMessageKey) {
-            record = new ProducerRecord(topic, key, value);
+            record = new ProducerRecord(topic, null, null, key, value, propagatedHeaders);
         } else {
-            record = new ProducerRecord(topic, value);
+            record = new ProducerRecord(topic, null, null, null, value, propagatedHeaders);
         }
         return Collections.singletonList(record).iterator();
     }
 
+    private List<Header> getPropagatedHeaders(Exchange exchange, HeaderFilterStrategy headerFilterStrategy) {
+        return exchange.getIn().getHeaders().entrySet().stream()
+                .filter(entry -> !headerFilterStrategy.applyFilterToExternalHeaders(entry.getKey(), entry.getValue(), exchange))
+                .map(entry -> new RecordHeader(entry.getKey(), getHeaderValue(entry.getValue())))
+                .collect(Collectors.toList());
+    }
+
+    private byte[] getHeaderValue(Object value) {
```

Comment:
Is there particular reason to keep String, Long, integer and byte[]?
can I send Double or Byte or Boolean[]?

Reply:
I've put most common types, so if you put unsupported one - value will be null.
The docs of kafka says that we need to bypass `byte[]` , but in camel headers we have `Object`, so I need somehow to know how to convert provided object to bytes.

Although I agree, that `Double` also good to have supported.

Reply:
if value type is not supported you just silently put null value -- there is no reason for the dev why this doesn't work: no log or exception... i don't think this is nice

Reply:
agree, added `warn` message 
+ added `Double` support